File: code-coaches/medium/divisible/divisible.rb
def is_divisible_by_all(num, dividers)
  dividers.each do |divider|
    return false if num % divider != 0
  end
  true
end

def main
  num = gets.chomp.to_i
  dividers = gets.chomp.split.map(&:to_i)

  if is_divisible_by_all(num, dividers)
    puts "divisible by all"
  else
    puts "not divisible by all"
  end
end

if __FILE__ == $0
  main
end

File: code-coaches/medium/divisible/Divisible.swift
func isDivisibleByAll(_ num: Int, _ dividers: [Int]) -> Bool {
    for divider in dividers {
        if num % divider != 0 {
            return false
        }
    }
    return true
}

func main() {
    let num = Int(readLine()!) ?? 0
	let dividers = readLine()!.split(separator: " ").map({ Int($0)! })

	if isDivisibleByAll(num, dividers) {
		print("divisible by all")
	} else {
		print("not divisible by all")
	}
}

main()
